src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Reflection/SignatureTypes.cs

```diff
@@ -48,7 +43,45 @@ public static IEnumerable<object[]> IsSignatureTypeTestData
                 yield return new object[] { typeof(List<>).MakeGenericType(sigType), true };
 
                 yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), typeof(int)), true };
+                yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), typeof(int)).GetGenericArguments()[0], false };
                 yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), sigType), true };
+                yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), sigType).GetGenericArguments()[0], true };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IsValueTypeTestData))]
+        public static void IsValueType(Type type, bool expected)
+        {
+            Assert.Equal(expected, type.IsValueType);
+        }
+
+        public static IEnumerable<object[]> IsValueTypeTestData
+        {
+            get
+            {
+                // These have normal generic argument types, not SignatureTypes. Using a SignatureType from MakeGenericMethodParameter()
+                // as the generic type argument throws NotSupportedException which is verified in MakeSignatureConstructedGenericType().
+                yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), typeof(int)).GetGenericArguments()[0], true };
```

Comment:
`Type.MakeGenericSignatureType(typeof(List<>), typeof(int)).GetGenericArguments()[0]` is going to be `typeof(int)`, so this test does not seem to be actually testing IsValueType property on signature types.

Should this rather be something like:

```
yield return new object[] { Type.MakeGenericSignatureType(typeof(List<>), typeof(int)), false };
yield return new object[] { Type.MakeGenericSignatureType(typeof(KeyValuePair<,>), typeof(string), typeof(object)), true };
```
?

Reply:
Correct this is not verifying signature types but is verifying that when a non-signature type is used as a generic parameter then IsValueType works as expected. The test with `GetGenericArguments()` is done in MakeSignatureConstructedGenericType() as mentioned in the comment.

I'm not sure if your feedback here came before or after I pushed a new commit that changed this and added the comment.

Reply:
My higher level concerns in both this and the other comment are:
- Missing test coverage for the code introduced in this PR. For example, if `SignatureConstructedGenericType.IsEnum` implementation is changed to return hardcoded `false`, no test is going to fail. These test gaps should be fixed. They are hard kind of hard to see because they are muddied by the unnecessary tests.
- Unnecessary test coverage for Types that are not signature types. It would be best to delete most of it and assume that the non-signature Types are tested elsewhere. This is pre-existing problem in these tests. I would not mind if you delete tests from this file that are not relevant to testing of signature types.

Reply:
I did remove newly added code that did not directly verify sig types.

Added tests for `IsEnum` for both true and false.

Also I found that the validation in MakeGenericSignatureType() did not check it the incoming type was generic or not, so I updated that and compared against MakeGenericType() which I also updated since the parameter names were different in the exception text (the `internal override` changed the parameter name from the `public abstract`). So in this case, I did add tests that are not sig-specific but it did catch this issue plus it's nice to see in one location.

Reply:
> Also I found that the validation in MakeGenericSignatureType() did not check it the incoming type was generic or not, so I updated that

This is technically a breaking change. Just pointing it out, I do not see a problem with it.

Reply:
I wonder if this breaking change meets the threshold to document it; I'll mark this breaking change for now and re-evaluate later as well as update the exception doc.